airgun/views/cloud_inventory.py

```diff
@@ -89,15 +89,24 @@ def child_widget_accessed(self, widget):
             self.click()
 
     def read(self, widget_names=None):
-        return {
+
+        final_dict = {
             'generating': self.generating.read(),
-            'uploading': self.uploading.read(),
+            'uploading': self.uploading.read() if self.uploading.is_displayed else None,
             'status': self.status,
-            'auto_update': self.parent.auto_update.read(),
+            'auto_update': (
+                self.parent.auto_update.read() if self.parent.auto_update.is_displayed else None
+            ),
             'obfuscate_hostnames': self.parent.obfuscate_hostnames.read(),
             'obfuscate_ips': self.parent.obfuscate_ips.read(),
             'exclude_packages': self.parent.exclude_packages.read(),
         }
+        # Remove 'uploading' if not displayed (based on SubscriptionConnectionSetting)
+        if not self.uploading.is_displayed:
+            final_dict.pop('uploading')
+        if not self.parent.auto_update.is_displayed:
+            final_dict.pop('auto_update')
+        return final_dict
```

Comment:
I'm not completely sure about the reason for the `pop`s, wouldn't this work similarly?
```
        final_dict = {
            'generating': self.generating.read(),
            'status': self.status,
            'obfuscate_hostnames': self.parent.obfuscate_hostnames.read(),
            'obfuscate_ips': self.parent.obfuscate_ips.read(),
            'exclude_packages': self.parent.exclude_packages.read(),
        }
        if self.uploading.is_displayed:
            final_dict['uploading'] = self.uploading.read()
        if self.parent.auto_update.is_displayed:
            final_dict['auto_update'] = self.auto_update.read()
```